scripts/install/lazylibrarian.sh

```diff
@@ -0,0 +1,162 @@
+
+##########################################################################
+# Functions
+##########################################################################
+
+function _dependencies() {
+    echo_info "Installing dependencies for $pretty_name..."
+
+    echo_progress_start "Creating $pretty_name venv"
+    mkdir -p "$venv_dir"
+    python3_venv "$master" "$app_name"
+    echo_progress_done
+
+    echo_progress_start "Installing python dependencies to venv"
+    # shellcheck disable=2154           # log variable is inherited from box itself.
+    "$venv_dir/bin/pip" install --upgrade pip >>"${log}" 2>&1 # Upgrade PIP
+    PIP='apprise pyOpenSSL'
+    # shellcheck disable=2086           # We want the $PIP variable to expand. So 2086's warning is invalid here.
+    "$venv_dir/bin/pip" install $PIP >>"${log}" 2>&1
+    chown -R "$master": "$venv_dir"
+    echo_progress_done
+    # TODO: https://lazylibrarian.gitlab.io/config_rtorrent/ says we would require libtorrent for rtorrent compatability
+}
+
+function _install() {
+    echo_progress_start "Installing $pretty_name..."
+    mkdir -p "$app_dir"
+    mkdir -p "$data_dir"
+    #    Git clone/extract LL wherever you like
+    # shellcheck disable=2154           # log variable is inherited from box itself.
+    git clone "https://gitlab.com/LazyLibrarian/LazyLibrarian.git" "$app_dir" >>"$log" 2>&1
+    if [[ ! -e "$app_dir/LazyLibrarian.py" ]]; then
+        ecoh_error "Git clone unsuccessful. Try running the box install again."
+        exit 1
+    fi
+    chown -R "$master": "$app_dir" # Change owner\group recursively for new dirs.
+    chown -R "$master": "$data_dir"
+    echo_progress_done
+}
+
+function _configure() {
+    #    Fill in all the config (see docs for full configuration)
+    echo_progress_start "Configuring $pretty_name..."
+    cat >"$data_dir/config.ini" <<EOF
+[General]
+http_root = /lazylibrarian
+
+[Git]
+auto_update = 1
+EOF
+    echo_progress_done
+}
+
+function _systemd() {
+    # CommandLine Options¶
+    #-d --daemon - Run the server as a daemon - not available on windows
+    #-q --quiet - Don't log to console
+    #-p --pidfile - Store the process id in this file
+    #--debug - Show debug log messages
+    #--nolaunch - Don't launch browser
+    #--update - Update to latest version from GitHub at start-up (only git installations)
+    #--port - Force webserver to listen on this port
+    #--datadir - Path to the datadir used to store database, config, cache - Default is to use program directory if this is unset
+    #--config - Path to config.ini - Default is to use datadir
+    #    Run "python LazyLibrarian.py -d" to start in daemon mode
+    echo_progress_start "Configuring $pretty_name systemd service..."
+    cat >"/etc/systemd/system/$app_name.service" <<EOF
+[Unit]
+Description=LazyLibrarian
+After=network.target
+
+[Service]
+User=$master
+Group=$master
+Type=simple
+ExecStart=$venv_dir/bin/python \
+$app_dir/LazyLibrarian.py \
+--datadir $data_dir
```

Comment:
I would make this one line and also maybe with `--nolaunch`

Reply:
Would we want to also make the pidfile just in case?

Reply:
Either --nolaunch or --daemon. I assume --daemon implies --nolaunch. Will fiddle with it a bit today.

PIDFILE makes sense to me. I left all those flags there so you would tell me which might be handy. ;-)

Can make it a one liner, but in practice they are identical. The service generated has a single line regardless if it's multiline here or not. I just find the code slightly easier to read this way. Up to you.

Reply:
Alright. Looks like all three will be needed.
`--nolaunch` because `--daemon` doesn't trigger `--nolaunch`
`--daemon` because `--pidfile` does nothing without it.
`--pidfile` for an easy outside reference to which PID the daemon is run on.

Reply:
Done